src/gui/dialogs/multiplayer/mp_create_game.hpp

```diff
@@ -126,6 +124,30 @@ class mp_create_game : public modal_dialog, private plugin_executor
 	void load_game_callback();
 
 	enum tab { TAB_GENERAL, TAB_OPTIONS, TAB_SETTINGS };
+
+	//
+	// mp_parameter defaults
+	//
+	void set_default_values();
+	bool force_lock_settings();
+
+	static std::string game_name_default();
```

Comment:
IMO it would be better to have these as part of mp_game_settings, if anything. No need to couple them to the UI.

Reply:
The issue is that determining the defaults needs to use:
* The create engine's current level (`create_engine_.current_level().data()`)
* The create engine's classification (`create_engine_.get_state().classification()`)

It also uses some methods in the `settings` namespace (`map_settings.*pp`) which documents itself as:
```
/**
 * Contains the general settings which have a default.
 * These values can be used to initialize the preferences,
 * but also to set map values if these aren't defined.
 */
```
so if the defaults methods shouldn't be in `mp_create_game`, then IMO they should go into `settings` and pass in whatever is needed.